File: TCDNew-master/src/ui.android/lib/models/dtos/PurchaseTaskLineStateDto.dart
class PurchaseTaskLineStateDto {
  DateTime expirationDate;
  int expirationDaysPlus;
  num qtyNormal;
  num qtyBroken;
  num get qtyFull => qtyNormal + qtyBroken;

  PurchaseTaskLineStateDto(this.expirationDate, this.expirationDaysPlus, this.qtyNormal, this.qtyBroken);

  factory PurchaseTaskLineStateDto.fromJson(Map<String, dynamic> json) {
    PurchaseTaskLineStateDto data;

    if (json != null) {
      data = PurchaseTaskLineStateDto(
          json['expirationDate'] != null
            ? DateTime.fromMillisecondsSinceEpoch(json['expirationDate'], isUtc: true)
            : null,
          json['expirationDaysPlus'] ?? 0,
          json['qtyNormal'] ?? 0.00,
          json['qtyBroken'] ?? 0.00
      );
    }

    return data;
  }

  toJson() {
    return {
      'expirationDate': expirationDate != null
        ? expirationDate.millisecondsSinceEpoch
        : null,
      'expirationDaysPlus': expirationDaysPlus,
      'qtyNormal': qtyNormal,
      'qtyBroken': qtyBroken,
    };
  }
}

File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Models/Dtos/PurchaseTaskLineStateDto.cs
using System;

namespace Cen.Wms.Client.Models.Dtos
{
    public class PurchaseTaskLineStateDto
    {
        public DateTime? ExpirationDate { get; set; }
        public long ExpirationDaysPlus { get; set; }
        public decimal QtyNormal { get; set; }
        public decimal QtyBroken { get; set; }

        public decimal QtyFull { get { return QtyNormal + QtyBroken; } }
    }
}
